### PR TITLE
Improvements to sync migrations

### PR DESCRIPTION
**Pull request by orta**
We were crashing when migrating between Folio versions (to an album sync version) when there were albums with no artworks empty. This protects against that.

### CODE REVIEW

File: Classes/Models/Generated/_AlbumDelete.h [hunk 1]
```diff
@@ -23,7 +23,7 @@ NS_ASSUME_NONNULL_BEGIN
 @interface _AlbumDelete : ARManagedObject
```
Comment (orta) [inline Classes/Models/Generated/_AlbumDelete.h]: These are all mogenerator updates

File: Classes/Sync/Albums/ARAlbumChangeUploader.m [hunk 2]
```diff
@@ -28,8 +28,13 @@ - (instancetype)initWithContext:(NSManagedObjectContext *)context
 
 - (void)operationTree:(DRBOperationTree *)tree objectsForObject:(NSString *)partnerID completion:(void (^)(NSArray *))completion
 {
```
Comment (orta) [inline Classes/Sync/Albums/ARAlbumChangeUploader.m]: back to code now

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Needs testing on a Phone if change is non-trivial</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by orta** [comment]
This should get merged BTW, as it's what is shipped.

**Pull request merged**